### PR TITLE
Fix retry to download local treeinfo file (#1270354)

### PR DESCRIPTION
**Pull request by jkonecny12**
When no repository is specified then we are trying /run/install/repo folder.

We were trying to download .treeinfo file from there and failed six times before giving up.
Now when .treeinfo got an error code which means local file can't be reached it will fail immediately.

*Related: rhbz#1270354*

*Reported-by: Peter Kotvan ``<pkotvan@redhat.com>``*

### CODE REVIEW

File: pyanaconda/packaging/__init__.py [hunk 1]
```diff
@@ -460,11 +460,22 @@ def _getTreeInfo(self, url, proxy_url, sslverify):
                 except URLGrabError as ex:
                     log.info("Error downloading 'treeinfo': %s", ex)
 
-                    # Don't retry if we got a 'HTTP Error 404' from both reponses
-                    # from curl (likely not to be a transient error), just move on.
-                    if "404" in str(e) and "404" in str(ex):
+                    # HTTP error
+                    if e.errno == 14 and ex.errno == 14:
```
Comment (rvykydal) [inline pyanaconda/packaging/__init__.py]: You could use something like
e.errno == ex.errno == 14
but it's a matter of taste.

**Comment by rvykydal** [comment]
Looks good to me.

File: pyanaconda/packaging/__init__.py (hunk 1 above)
Comment (jkonecny12) [inline pyanaconda/packaging/__init__.py]: Why not, it's smaller that way.

**Comment by jkonecny12** [comment]
Pushed, thank you.

**Pull request closed without merging**